bundles/orbisgis-omanager/src/main/java/org/orbisgis/omanager/ui/ActionDeploy.java

```diff
@@ -111,20 +112,22 @@ private void deployBundle(Resolver resolver) {
             }
             if (DependencyMessageDialog.showModal(SwingUtilities.getWindowAncestor(frame),
                     I18N.tr("Dependencies downloading"), resourcesNames.toString(), dependencyStr.toString())
-                    .equals(DependencyMessageDialog.CHOICE.OK)) {
-
-                DeployBundleSwingWorker worker = new DeployBundleSwingWorker(resolver);
-                ServiceReference<ExecutorService> executorServiceReference = bundleContext.getServiceReference(ExecutorService.class);
-                ExecutorService executorService = null;
-                if(executorServiceReference != null){
-                    executorService = bundleContext.getService(executorServiceReference);
-                }
-                if(executorService != null){
-                    executorService.execute(worker);
-                }
-                else{
-                    worker.execute();
-                }
+                    .equals(DependencyMessageDialog.CHOICE.CANCEL)) {
+                deployBundle = false;
+            }
+        }
+        if(deployBundle){
+            DeployBundleSwingWorker worker = new DeployBundleSwingWorker(resolver);
+            ServiceReference<ExecutorService> executorServiceReference = bundleContext.getServiceReference(ExecutorService.class);
```

Comment:
Yes because calling bundleContext.getServiceReference has to be avoided because OSGi must handle the services lifcycle itself. (less complicated things as OrbisGIS grown)